mod.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020-2021 the Deno authors. All rights reserved. MIT license.
+
+import { doc as jsDoc } from "./lib/deno_doc.js";
+import type { DocNode } from "./lib/types.d.ts";
+import { load as defaultLoad } from "https://deno.land/x/deno_graph@0.2.0/lib/loader.ts";
+import type { LoadResponse } from "https://deno.land/x/deno_graph@0.2.0/mod.ts";
+
+export type {
+  DocNode,
+  DocNodeClass,
+  DocNodeEnum,
+  DocNodeFunction,
+  DocNodeImport,
+  DocNodeInterface,
+  DocNodeKind,
+  DocNodeNamespace,
+  DocNodeTypeAlias,
+  DocNodeVariable,
+} from "./lib/types.d.ts";
+
+export type { LoadResponse } from "https://deno.land/x/deno_graph@0.2.0/mod.ts";
+
+export interface DocOptions {
+  /**
+   * An optional callback that is called with the URL string of the resource to
+   * be loaded and a flag indicating if the module was required dynamically. The
+   * callback should resolve with a `LoadResponse` or `undefined` if the module
+   * is not found. If there are other errors encountered, a rejected promise
+   * should be returned.
+   *
+   * This defaults to a load function which will use `fetch()` and
+   * `Deno.readFile()` to load modules, and requires the appropriate permissions
+   * to function. If the permissions are note available at startup, the default
+   * function will prompt for them.
+   *
+   * @param specifier The URL string of the resource to be loaded and resolved
+   * @param isDynamic A flag that indicates if the module was being loaded
+   *                  dynamically
+   */
+  load?(
+    specifier: string,
+    isDynamic: boolean,
+  ): Promise<LoadResponse | undefined>;
+  /** An optional callback that allows the default resolution logic of the
+   * module graph to be "overridden". This is intended to allow items like an
+   * import map to be used with the module graph. The callback takes the string
+   * of the module specifier from the referrer and the string URL of the
+   * referrer. The callback then returns a resolved URL string specifier. */
+  resolve?(specifier: string, referrer: string): string;
```

Comment:
Question here: in internal Deno infra this function has also `is_main` param which tells is the resolved file is the entry point - this is useful/required when used in combination with import map (which shouldn't be applied to the entry point, but should for all other files). Should we support it here?

Reply:
I think it would be more a question for `deno_graph`, but, I would say that is an externality that shouldn't be a part of the API.  If there is additional contextual information, like what the root module is, that should be applied, that should be communicated to the resolver some other way.  Your specific scenario would be that the root specifier provided to create the module graph would need to be also passed to the import map.